W3ChampionsStatisticService/Matches/MatchRepository.cs

```diff
@@ -176,12 +175,24 @@ public async Task<int> GetFloIdFromId(string gameId)
         return (match == null || match.FloMatchId == null) ? 0 : match.FloMatchId.Value;
     }
 
-    public Task<long> Count(
-        int season,
-        GameMode gameMode)
+    public Task<long> Count(int season, GameMode gameMode, HeroType hero = HeroType.AllFilter)
+    {
+        var filter = GetLoadFilter(season, gameMode, hero);
+        return CreateCollection<Matchup>().CountDocumentsAsync(filter);
+    }
+
+    private FilterDefinition<Matchup> GetLoadFilter(int season, GameMode gameMode, HeroType hero = HeroType.AllFilter)
     {
-        return CreateCollection<Matchup>().CountDocumentsAsync(m =>
-                gameMode == m.GameMode && m.Season == season);
+        var builder = Builders<Matchup>.Filter;
+        var filter = builder.Eq(m => m.GameMode, gameMode) & builder.Eq(m => m.Season, season);
+
+        if (hero != HeroType.AllFilter && hero != HeroType.Unknown)
+        {
+            var heroFilter = builder.Where(m => m.Teams.Any(t => t.Players.Any(p => p.Heroes.Any(h => h.Id == hero))));
```

Comment:
I think it should probably match only the first hero in the list (vast majority of the time you'd want to search for opening hero for build order stuff, I think) but I'm not sure if the game is reliable about keeping the list in order they were trained

Reply:
Yeah, I was thinking about that. Not sure if the list we get is in order or not, but I guess we can just assume it is for now and see how it goes. Could also filter on highest level but would need adding an index on that as well I think which probably isn't worth it